File: API/src/dtos/comment.dto.ts
export class CommentDto {
  _id: string = "";
  publisherId: string = "";
  publisherUsername: string = "";
  content: string = "";
  createdAt: number = 0;

  public static mapFrom(comment: any): CommentDto {
    return new CommentDto(comment);
  }

  private constructor(comment?: any) {
    this._id = comment === null ? "" : comment._id;
    this.publisherId = comment === null ? "" : comment.userId;
    this.content = comment === null ? "" : comment.content;
    this.createdAt = comment === null ? 0 : comment.createdAt;
  }
}

File: BadcourtAPI/SharedKernel/DTOs/CommentDto.cs
namespace SharedKernel.DTOs;

public class CommentDto
{
    public string Id { get; set; } = null!;
    public Guid PublisherId { get; set; }
    public string PostId { get; set; } = null!;
    public string PublisherUsername { get; set; } = string.Empty;
    public string PublisherImageUrl { get; set; } = string.Empty;
    public string PublisherState { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public IEnumerable<FileDto> Resources { get; set; } = [];
    public int LikesCount { get; set; }
    public bool IsLiked { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
